code/go/0chain.net/smartcontract/dbs/event/delegate_pool.go

```diff
@@ -0,0 +1,148 @@
+package event
+
+import (
+	"errors"
+	"fmt"
+
+	"0chain.net/smartcontract/dbs"
+	"gorm.io/gorm"
+)
+
+type DelegatePool struct {
+	gorm.Model
+
+	PoolID       string `json:"pool_id"`
+	ProviderType int    `json:"provider_type"`
+	ProviderID   string `json:"provider_id"`
+	DelegateID   string `json:"delegate_id"`
+
+	Balance      int64 `json:"balance"`
+	Reward       int64 `json:"reward"`
+	TotalReward  int64 `json:"total_reward"`
+	TotalPenalty int64 `json:"total_penalty"`
+	Status       int   `json:"status"`
+	RoundCreated int64 `json:"round_created"`
+}
+
+func (edb *EventDb) overwriteDelegatePool(sp DelegatePool) error {
+	result := edb.Store.Get().
+		Model(&DelegatePool{}).
+		Where(&DelegatePool{
+			PoolID:       sp.PoolID,
+			ProviderType: sp.ProviderType,
+		}).Updates(map[string]interface{}{
+		"delegate_id":   sp.DelegateID,
+		"provider_type": sp.ProviderType,
+		"provider_id":   sp.ProviderID,
+		"pool_id":       sp.PoolID,
+		"balance":       sp.Balance,
+		"reward":        sp.Reward,
+		"total_reward":  sp.TotalReward,
+		"total_penalty": sp.TotalPenalty,
+		"status":        sp.Status,
+		"round_created": sp.RoundCreated,
+	})
+	return result.Error
+}
+
+func (sp *DelegatePool) exists(edb *EventDb) (bool, error) {
+	var dp DelegatePool
+	result := edb.Store.Get().Model(&DelegatePool{}).Where(&DelegatePool{
+		ProviderType: sp.ProviderType,
+		PoolID:       sp.PoolID,
+	}).Take(&dp)
+	if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+		return false, nil
+	}
+	if result.Error != nil {
+		return false, fmt.Errorf("failed to check Curator existence %v, error %v",
+			dp, result.Error)
+	}
+	return true, nil
+}
+
+func (edb *EventDb) updateReward(reward int64, dp DelegatePool) error {
+	dpu := dbs.NewDelegatePoolUpdate(dp.PoolID, dp.ProviderID, dp.ProviderType)
+
+	if dp.ProviderType == int(dbs.Blobber) && reward < 0 {
+		dpu.Updates["total_penalty"] = dp.TotalPenalty - reward
+	} else {
+		dpu.Updates["reward"] = dp.Reward + reward
```

Comment:
what's the difference between reward and total_reward?

Reply:
`reward` is the unclaimed reward, that can be transferred to the user's account with a claim_reward transaction.
`total_reward` is a summary statistic holding the total of all rewards paid to this particular delegate pool.